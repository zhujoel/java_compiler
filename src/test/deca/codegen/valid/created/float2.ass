; start main program
; Class Declaration
	LOAD #null, R0
	STORE R0, 1(GB)
	LOAD code.Object.equals, R0
	STORE R0, 2(GB)
; Main Function
; Beginning of main instructions:
; Test pour savoir si la pile est pleine
	TSTO #3
	BOV pile_pleine
	ADDSP #3
; Declaration des variables
	LOAD #0x1.0p0, R2
	STORE R2, 3(GB)
; Instructions
; +
	LOAD 3(GB), R2
	LOAD #0x1.4p1, R3
	ADD R2, R3
	BOV debordement
	LOAD R3, R1
	WINT
	HALT
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithm√©tique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
