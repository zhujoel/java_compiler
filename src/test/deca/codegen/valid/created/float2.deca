{
    float a = 1.0;
    print(a + 2.5);
}
