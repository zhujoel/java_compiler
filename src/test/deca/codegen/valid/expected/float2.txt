3.50000e+00
